src/main/java/com/epam/training/gen/ai/configuration/Config.java

```diff
@@ -48,12 +54,20 @@ public ChatCompletionService chatCompletionService(OpenAIAsyncClient openAIAsync
     }
 
     @Bean
-    public Kernel kernel(ChatCompletionService chatCompletionService) {
+    public Kernel kernel(ChatCompletionService chatCompletionService,
+                         KernelPlugin creatorInfoPlugin) {
```

Comment:
Only one question,When you change the model using the request parameters, does everything work as you expect? I believe you might need to create a new chatCompletionService for each different model and connect it to the kernel accordingly. It seems like the default model is being used all the time, no matter what is specified in the request.

Reply:
Yes, it works as expected.

Reply:
https://github.com/bodik777/first-ai-project/blob/task_4/Task3_ChatGPT.txt
https://github.com/bodik777/first-ai-project/blob/task_4/Task3_Gemini.txt
here are results of testing